### PR TITLE
[FIX] Broken Omnichannel>Agents page

### PR DESCRIPTION
**Pull request by tiagoevanp**

### CODE REVIEW

File: apps/meteor/client/views/omnichannel/agents/AgentsRoute.tsx [hunk 1]
```diff
@@ -63,9 +63,9 @@ const AgentsRoute: FC = () => {
 	const context = useRouteParameter('context');
 	const id = useRouteParameter('id');
 
-	if (!id) {
-		throw new Error('Agent id is required');
-	}
+	// if (!id) {
+	// 	throw new Error('Agent id is required');
+	// }
```
Comment (KevLehman) [inline apps/meteor/client/views/omnichannel/agents/AgentsRoute.tsx:68]: 👀

Comment (tiagoevanp) [reply apps/meteor/client/views/omnichannel/agents/AgentsRoute.tsx:68]: ouch!

File: apps/meteor/client/components/FilterByText.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ const FilterByText: FC<FilterByTextProps> = ({ placeholder, onChange: setFilter,
 	}, []);
 
 	useEffect(() => {
-		setFilter({ text });
```
Comment (MartinSchoeler) [inline apps/meteor/client/components/FilterByText.tsx:30]: Isn’t `setFilter` a required prop? It should never be undefined, if it is happening, we should fix the place that is not setting the prop

Or make it optional.

**Review by MartinSchoeler — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/omnichannel/agents/AgentsRoute.tsx [hunk 3]
```diff
@@ -180,8 +176,8 @@ const AgentsRoute: FC = () => {
 					<VerticalBar.Close onClick={handleVerticalBarCloseButtonClick} />
 				</VerticalBar.Header>
 
-				{context === 'edit' && <AgentEditWithData uid={id} reload={reload} />}
-				{context === 'info' && (
+				{context === 'edit' && id && <AgentEditWithData uid={id} reload={reload} />}
```
Comment (ggazzo) [inline apps/meteor/client/views/omnichannel/agents/AgentsRoute.tsx:179]:
```suggestion
				{context === 'edit' && (id ? <AgentEditWithData uid={id} reload={reload} />: <SomeErrorPage />}
```

Comment (tiagoevanp) [reply apps/meteor/client/views/omnichannel/agents/AgentsRoute.tsx:179]: ok

**Review by MartinSchoeler — CHANGES_REQUESTED** [review]

**Comment by murtaza98** [comment]
@tiagoevanp Just a suggestion to add a basic test case for this page to verify if this page is getting loaded or not - just so we can avoid this type of issue in future

**Comment by tiagoevanp** [comment]
> 

Yes, thanks for the comment Murtaza. I know that we have different teams working on tests outside of our main team at rocket.chat... Probably this case will be covered by them.

**Review by MartinSchoeler — APPROVED** [review]

**Pull request merged**